oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/jackson2/OAuth2AuthenticationTokenMixinTests.java

```diff
@@ -194,7 +194,8 @@ private static String asJson(DefaultOAuth2User oauth2User) {
 				"      \"@class\": \"java.util.Collections$UnmodifiableMap\",\n" +
 				"      \"username\": \"user\"\n" +
 				"    },\n" +
-				"    \"nameAttributeKey\": \"username\"\n" +
+				"    \"nameAttributeKey\": \"username\",\n" +
```

Comment:
I don't think this is representative of how the JSON will look (no more `nameAttributeKey` property). 

The tests as a whole should confirm two things:

1. Values serialized with no `name` property (given that it is new) should still deserialize correctly
2. Values with no `nameAttributeKey` property (given that it is being removed) should still deserialize correctly

I believe this means that there will need to be a change to `DefaultOAuth2UserMixin`; it may need to use a deserializer, like `ClientRegistrationMixin` does. If it contains `nameAttributeKey`, use the deprecated constructor; otherwise, use the new constructor.

Reply:
@jzheaux, here after implementing deserializer for `DefaultOAuth2UserMixin`, I had to implement another deserializer `DefaultOidcUserMixin`. seems reasonable ?? just checking if I am in the right direction

Reply:
@jzheaux, just to make sure I am going the right direction, working on this I had to create 2 deserializers one for `DefaultOAuth2UserMixin` and another for `DefaultOidcUserMixin` seems reasonable ?